### PR TITLE
Add queue to SHMEM

### PR DESCRIPTION
**Pull request by EtK2000**
An attempt to mitigate the packet loss

### CODE REVIEW

File: orkaudio/audiocaptureplugins/voip/VoIp.cpp [hunk 1]
```diff
@@ -1004,6 +1015,28 @@ void HandlePacket(u_char *param, const struct pcap_pkthdr *header, const u_char
...
+    SetThreadName("PacketThreadHandler");
+
+    // lower our priority, TODO: this might be too low...
+    pthread_setschedprio(pthread_self(), sched_get_priority_min(sched_getscheduler(getpid())));
+
+    PcapPacketData p;
+    for (;;) {
+
+        // whilst backlog has packets handle them
+        while (voip_backlog.pop(p)) {
```
Comment (arikhalperin) [inline orkaudio/audiocaptureplugins/voip/VoIp.cpp:1029]: What about PCapPacketData from the queue? How is it freed?

Comment (arikhalperin) [reply orkaudio/audiocaptureplugins/voip/VoIp.cpp:1029]: Is the pop freeing the pointer which holds it? Is this what it does? Copy the data to p and free the pointer?

File: orkaudio/audiocaptureplugins/voip/VoIp.cpp [hunk 2]
```diff
@@ -1766,10 +1799,10 @@ void VoIp::LoadPartyMaps()
 	struct stat finfo;
 	CStdString filename = LOCAL_PARTY_MAP_FILE;
 	int rt = -1;
-	if(stat((PCSTR)filename.c_str(), &finfo) != 0)
+	if(stat(filename.c_str(), &finfo) != 0)
```
Comment (arikhalperin) [inline orkaudio/audiocaptureplugins/voip/VoIp.cpp:1802]: Why did you do this change?

Comment (EtK2000) [reply orkaudio/audiocaptureplugins/voip/VoIp.cpp:1802]: The cast does nothing, so I just removed it.
string::c_str() returns a `const char *` already.

File: orkaudio/audiocaptureplugins/voip/VoIp.cpp (hunk 1 above)
Comment (EtK2000) [reply orkaudio/audiocaptureplugins/voip/VoIp.cpp:1029]: Hmm...
I'll write test code quickly.
I was under the impression of a single copy construction, but you're probably right about there being 2.

**Pull request merged**